architecture/image-controller.md

```diff
@@ -1,54 +1,152 @@
 
-- **Setup image repository**: Image controller creates an image repository for the Component CR in a remote image registry as well as a robot account which is specific to that repository for image push. A Kubernetes Secret object is also created with that robot account token in order to make it available for build PipelineRun.
+- **Setup image repository**: create an image repository and a robot accounts
+  which are specific to that repository for image push and pull.
+  A Kubernetes Secret objects are also created with push and pull robot account tokens,
+  in order to make it available for build PipelineRun via service account.
 
-- **Modify visibility**: Image controller is able to switch an image repository's visibility between public and private.
+- **Modify repository visibility**: switch an image repository's visibility between public and private.
 
-- **Cleanup**: When a Component CR is requested to be deleted, image controller will remove component's image repository and robot account from the remote registry. The Kubernetes Secret will be removed along with the Component CR eventually due to the ownership established between them.
+- **Rotating credentials for repository**: rotate repository credentials, and update relevant secrets.
+
+- **Verify and fix secrets linked to ServiceAccount**: verify and fix linking of secrets to ServiceAccount.
+
+- **Cleanup**: When a Component CR is requested to be deleted, image controller will remove
+  component's image repository and robot account from the remote registry
+  (it is possible to skip repository removal).
+  The Kubernetes Secret will be removed along with the Component CR eventually due to the
+  ownership established between them.
 
 # Dependencies
 
-Image controller does not depend on other Konflux services, but a remote image registry. Konflux services are able to use the resources prepared by image controller, e.g. Build Service makes the Secret available to every build PipelineRun of a component for image push.
+Image controller does not depend on other Konflux services, but a remote image registry.
+Konflux services are able to use the resources prepared by image controller,
+e.g. ServiceAccount with linked Secrets is available to every build PipelineRun of a component
+for pushing image.
 
 # Interface
 
 ## ImageRepository CR
 
 The ImageRepository CR is the interface to interact with image controller to create and manage image repositories in a registry.
 
-To create an image repository for a Component, apply this YAML code:
+### To create an general purpose image repository, apply this YAML code:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ImageRepository
+metadata:
+    name: imagerepository-for-component-sample
+    namespace: test-ns
+```
+As a result, a public image repository `quay.io/my-org/test-ns/imagerepository-sample` will be created,
+based on `$DEFAULT_REGISTRY_ORG/$USER_NAMESPACE/$IMAGE_REPOSITORY_NAME`.
```

Comment:
I don't really get this part. Is it that:
- `$USER_NAMESPACE` is `.metadata.namespace`
- `$IMAGE_REPOSITORY_NAME` is `.metadata.name`
- `$DEFAULT_REGISTRY_ORG` is configured somewhere else

?

Reply:
yes

Reply:
Do you think it would be worth clarifying, and maybe mentioning where the default is configured?